chall05/jtrancos.c

```diff
@@ -0,0 +1,29 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   jtrancos.c                                         :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: jtrancos <marvin@42.fr>                    +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2020/04/16 13:01:08 by jtrancos          #+#    #+#             */
+/*   Updated: 2020/04/16 13:36:56 by jtrancos         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <stdio.h>
+
+int	ft_ie_except_after_c(char* str)
+{
+	int i;
+
+	i = 0;
+	while (str[i])
+	{
+		if (str[i] == 'e' && str[i + 1] == 'i' && str[i - 1] != 'c')
```

Comment:
What if `e` is the first character? You'll just go out of the string and crash  
You should try to compile with the flag `-fsanitize=address` to be more aware of such problems.

Reply:
It could be solved by adding the condition `if (i != 0)` before the `if` right?